gensim/downloader.py

```diff
@@ -0,0 +1,274 @@
+
+user_dir = os.path.expanduser('~')
+base_dir = os.path.join(user_dir, 'gensim-data')
+logger = logging.getLogger('gensim.api')
+
+
+def progress(chunks_downloaded, chunk_size, total_size):
+    r"""Create and update the progress bar.
+
+    Parameters
+    ----------
+    chunks_downloaded : int
+        Number of chunks of data that have been downloaded
+    chunks_size : int
+        Size of each chunk of data.
+    total_size : int
+        Total size of the dataset/model
+
+    References
+    ----------
+    [1] https://gist.github.com/vladignatyev/06860ec2040cb497f0f3
+
+    """
+    bar_len = 50
+    size_downloaded = float(chunks_downloaded * chunk_size)
+    filled_len = int(math.floor((bar_len * size_downloaded) / total_size))
+    percent_downloaded = round((size_downloaded * 100) / total_size, 1)
+    bar = '=' * filled_len + '-' * (bar_len - filled_len)
+    sys.stdout.write('[%s] %s%s %s/%sMB downloaded\r' % (bar, percent_downloaded, "%", round(size_downloaded / (1024 * 1024), 1), round(float(total_size) / (1024 * 1024), 1)))
+    sys.stdout.flush()
+
+
+def _create_base_dir():
+    r"""Create the gensim-data directory in home directory, if it has not been already created.
+    Raises
+    ------
+    File Exists Error
+        Give this exception when a file gensim-data already exists in the home directory.
+    Exception
+        An exception is raised when read/write permissions are not available
+    """
+    if not os.path.isdir(base_dir):
+        try:
+            logger.info("Creating %s", base_dir)
+            os.makedirs(base_dir)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                raise Exception(
+                    "Not able to create folder gensim-data in {}. File gensim-data "
+                    "exists in the direcory already.".format(user_dir))
+            else:
+                raise Exception(
+                    "Can't create {}. Make sure you have the read/write permissions "
+                    "to the directory or you can try creating the folder manually"
+                    .format(base_dir))
+
+
+def _calculate_md5_checksum(tar_file):
+    r"""Calculate the checksum of the given tar.gz file.
+    Parameters
+    ----------
+    tar_file : str
+        Path to the tar.gz file that contains the downloaded dataset/model.
+
+    Returns
+    -------
+    str
+        Return the calculated checksum of the tsr.gz file.
+
+    """
+    hash_md5 = hashlib.md5()
+    with open(tar_file, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def info(name=None):
+    r"""Return the information related to model/dataset.
+
+    If name is supplied, then information related to the given dataset/model will be returned. Otherwise detailed information of all model/datasets will be returned.
+
+    Parameters
+    ----------
+    name : {None, data name}, optional
```

Comment:
`name : str or None, optional` is the right way. Also try to write description after every argument.